src/devices/watts_thermostat.c

```diff
@@ -0,0 +1,199 @@
+
+This code is based on a slightly older OEM system created by ADEV in France which
+later merged with Watts. The closest thing currently available seems to be
+https://wattswater.eu/catalog/regulation-and-control/radio-wfht-thermostats/electronic-room-thermostat-with-rf-control-wfht-rf-basic/,
+but it is not known whether they are protocol compatible.
+
+https://github.com/merbanan/rtl_433/issues/2230
+
+Analyzer output:
+    Analyzing pulses...
+    Total count:   55,  width: 337.75 ms            (675502 S)
+    Pulse width distribution:
+     [ 0] count:    1,  width: 291096 us [291096;291096]    (582191 S)
+     [ 1] count:   26,  width:  630 us [630;640]    (1260 S)
+     [ 2] count:   28,  width:  288 us [286;302]    ( 576 S)
+    Gap width distribution:
+     [ 0] count:   26,  width:  233 us [232;234]    ( 466 S)
+     [ 1] count:   28,  width:  576 us [576;576]    (1151 S)
+    Pulse period distribution:
+     [ 0] count:    1,  width: 291328 us [291328;291328]    (582657 S)
+     [ 1] count:   15,  width: 1206 us [1205;1216]  (2412 S)
+     [ 2] count:   14,  width:  522 us [520;530]    (1043 S)
+     [ 3] count:   24,  width:  864 us [862;877]    (1727 S)
+    Pulse timing distribution:
+     [ 0] count:    1,  width: 291096 us [291096;291096]    (582191 S)
+     [ 1] count:   54,  width:  602 us [576;640]    (1204 S)
+     [ 2] count:   54,  width:  262 us [232;302]    ( 523 S)
+     [ 3] count:    1,  width: 100000 us [100000;100000]    (200001 S)
+    Level estimates [high, low]:  15961,    150
+    RSSI: -0.1 dB SNR: 20.3 dB Noise: -20.4 dB
+    Frequency offsets [F1, F2]:   17869,      0     (+545.3 kHz, +0.0 kHz)
+    Guessing modulation: Pulse Width Modulation with sync/delimiter
+    view at https://triq.org/pdv/#AAB104FFFF025A0105FFFF8291A291A1A291A29291A29291A291A1A1A2929291A29291A1A1A1A1A29291A291A1A1A1A291A1A29291A1A29291A29291A1A2929291A355
+    Attempting demodulation... short_width: 288, long_width: 630, reset_limit: 577, sync_width: 291096
+    Use a flex decoder with -X 'n=name,m=OOK_PWM,s=288,l=630,r=577,g=0,t=0,y=291096'
+    pulse_slicer_pwm(): Analyzer Device
+    bitbuffer:: Number of rows: 1
+    [00] {54} 5a 4b 89 f2 f6 64 c4
+
+
+Data Layout:
+
+    10100101   1011010001110110   1000   100100001   000011000   10101011
+    preamble   id                 flags  temp         setpoint   chksum
+
+    All fields need reflection, possibly easier to just reverse the whole
+    row first. The only flag found is PAIRING (0b0001). Chksum seems to be
+    additive.
+
+Raw data:
+
+    {54}5ab24971f79994
+    {54}5ab24971f79994
+    {54}5ab249f1f79b94
+    {54}5ab249f1f79b94
+    {54}5ab249f9f79854
+    {54}5ab249f5f79a54
+    {54}5ab249f68f998c
+    {54}5ab249f98f9a4c
+    {54}5ab249f58b9a4c
+    {54}5ab249fb8f9acc
+
+    https://tinyurl.com/2z5jtfuu
+
+    Format string:
+    ID:^16d FLAGS:^4b TEMP:^9d SETP:^9d CHK:^8d
+
+Decoded example:
+
+    ID:28205 FLAGS:0001 TEMP:265 SETP:048 CHK:214
```

Comment:
This isn't really like normal decoders, which describe the format and modulation and have a stylized comment with nibbles.    As such it is harder to understand.

Reply:
Please point me to a good example and I can try to follow that. The tinyurl points to a bitbench where those parameters are used (although they might not match exaclty). I took an existing decoder and adapted it so I didn't know exactly what to put where.

Reply:
Might be somewhat outdated but https://github.com/merbanan/rtl_433/blob/master/src/devices/new_template.c